internal/cmd/upgradecmd.go

```diff
@@ -634,3 +651,66 @@ func getReleaseAssetByName(rr *github.RepositoryRelease, name string) *github.Re
 	}
 	return nil
 }
+
+type InstallBehavior struct {
+	PortablePackageUserRoot    string `json:"portablePackageUserRoot"`
+	PortablePackageMachineRoot string `json:"portablePackageMachineRoot"`
+	DefaultInstallRoot         string `json:"defaultInstallRoot"`
+}
+
+func (ib *InstallBehavior) Values() []string {
+	return []string{
+		ib.PortablePackageUserRoot,
+		ib.PortablePackageMachineRoot,
+		ib.DefaultInstallRoot,
+	}
+}
+
+type WinGetSettings struct {
+	InstallBehavior InstallBehavior `json:"installBehavior"`
+}
+
+// isWinGetInstall determines if executableAbsPath contains a WinGet installation path.
+func isWinGetInstall(fileSystem vfs.Stater, executableAbsPath string) (bool, error) {
+	realExecutableAbsPath := executableAbsPath
+	fi, err := os.Lstat(executableAbsPath)
+	if err != nil {
+		return false, err
+	}
+	if fi.Mode().Type() == os.ModeSymlink {
+		realExecutableAbsPath, err = os.Readlink(executableAbsPath)
+		if err != nil {
+			return false, err
+		}
+	}
+	winGetSettings := WinGetSettings{
+		InstallBehavior: InstallBehavior{
+			PortablePackageUserRoot:    os.ExpandEnv("${LOCALAPPDATA}\\Microsoft\\WinGet\\Packages\\"),
```

Comment:
Quick tip: You can use backquoted strings to avoid having to escape all the backslashes, e.g.

```go
os.ExpandEnv(`${LOCALAPPDATA}\Microsoft\WinGet\Packages\`)
```